File: leakybucket.py
import time
import random
bucketSize =512
def bktInput(size,rate):
    if size>bucketSize:
        print("\n Bucket Overflow")
    else:
        time.sleep(1)
        while size>rate:
            print("{} bytes sent".format(rate))
            size -=rate
            time.sleep(1)
        if size >0:
            print(" Last {} bytes sent".format(size))
        print("\n Bucket output successful")

outputRate = int(input("Enter the ouput rate : "))
for i in range(5):
    print(random.randint(0,1000))
    packetSize = random.randint(0,1000)
    print("\n Packet Number : {}  \n Packet Size : {} ".format((i+1),packetSize))
    bktInput(packetSize,outputRate)

File: leakybucket.cpp
#include<iostream>
#include<unistd.h>
#include<stdlib.h>
#define bucketSize 512
using namespace std;

void bktInput(int a,int b)
{
	if(a>bucketSize)
		cout<<"\nBucket overflow";
	else
	{
		usleep(500);
		while(a>b)
		{
			cout<<"\n"<<b<<"bytes outputted.";
			a-=b;
			usleep(500);
		}
		if(a>0)
			cout<<"\n Last"<<a<<"bytes sent\t";
		cout<<"\n bucket output successful";
	}
}

int main()
{
	int op,pktSize;
	cout<<"enter output rate:";
	cin>>op;
	for(int i=1;i<=5;i++)
	{
		usleep(rand()%1000);
		pktSize=rand()%1000;
		cout<<"\nPacket no"<<i<<"\tPacket size"<<pktSize;
		bktInput(pktSize,op);
	}
}